src/diffpy/srmise/tests/test_dataclusters.py

```diff
@@ -44,13 +44,32 @@ def test___eq__():
                 "y": np.array([3, 2, 1]),
                 "res": 4,
             },
-            DataClusters(np.array([1, 2, 3]), np.array([3, 2, 1]), 4),
+            {
+                "x": np.array([1, 2, 3]),
+                "y": np.array([3, 2, 1]),
+                "res": 4,
+                "data_order": [2, 1, 0],
+                "clusters": np.array([[0, 0]]),
+                "current_idx": 2,
+                "lastpoint_idx": 0,
+                "INIT": 0,
+                "READY": 1,
+                "CLUSTERING": 2,
+                "DONE": 3,
+                "lastcluster_idx": None,
+                "status": 1,
+            }
         ),
     ],
 )
 def test_set_data(inputs, expected):
     actual = DataClusters(x=inputs["x"], y=inputs["y"], res=inputs["res"])
```

Comment:
this is better.  Strinctly this still doesn't test set_data alone.  It tests the object constructor.  I think this is ok, but we may want to make clear and set_data as private functions.  Then we don't need tests (or docstrings in prinicple) for them and we just test the constructor (the `__init__`).

Whether or not to do this depends where else these functions rae used.  Do we want to make them available to users to use, or are they just being used in __init__ alone or in __init__ and a few other places.

These are small things, but once we touch the code we want to leave it better than when we arrived, and it is also a good learning experience.....

Could you look how these two functions are used and we can decide.  If we make them private functions we can leave this test as it is but just change its name.

Reply:
> this is better. Strinctly this still doesn't test set_data alone. It tests the object constructor. I think this is ok, but we may want to make clear and set_data as private functions. Then we don't need tests (or docstrings in prinicple) for them and we just test the constructor (the `__init__`).
> 
> Whether or not to do this depends where else these functions rae used. Do we want to make them available to users to use, or are they just being used in **init** alone or in **init** and a few other places.
> 
> These are small things, but once we touch the code we want to leave it better than when we arrived, and it is also a good learning experience.....
> 
> Could you look how these two functions are used and we can decide. If we make them private functions we can leave this test as it is but just change its name.

I'm certain that these two functions are only used in the constructor. It should be OK to change them into private functions.

Reply:
super, let's do this.  Just 

- change the name of this test to something like `test_DataClusters_constructor` since this is what it does anyway.
- add an underscore to the beginning of the clear and set_data functions. You can leave the docstring since we already wrote it.
- revisit the test_clear tests.  We want to remove this test, but let's make sure that this behavior is being tested in the constructor test, so move over anything we need from there.